src/modules/users/Config.cpp

```diff
@@ -337,8 +372,16 @@ Config::setFullName( const QString& name )
         emit fullNameChanged( name );
 
         // Build login and hostname, if needed
-        QRegExp rx( "[^a-zA-Z0-9 ]", Qt::CaseInsensitive );
-        QString cleanName = CalamaresUtils::removeDiacritics( name ).toLower().replace( rx, " " ).simplified();
+        static QRegExp rx( "[^a-zA-Z0-9 ]", Qt::CaseInsensitive );
+#ifdef HAVE_ICU
+        QString cleanName = transliterate(name);
+        cleanName.replace("'", "");
+#else
+        QString cleanName = name;
+#endif
+        cleanName = CalamaresUtils::removeDiacritics( cleanName ).replace( rx, " " ).toLower().simplified();
```

Comment:
If you introduce a dummy `transliterate()` then the flow becomes clearer and more consistent:
```
QString cleanName = transliterate(name).replace("'","");
cleanName = // the diacritics bits
```
or maybe in three lines, so you can see the three steps:
 - transliterate
 - remove punctuation (possibly introduced by transliteration)
 - remove diacritics and split on non-alphanumerics